controllers/drplacementcontrol_controller.go

```diff
@@ -1249,74 +1346,88 @@ func (d *DRPCInstance) updateUserPlacementRuleStatus(status plrv1.PlacementRuleS
-	}
+	if err := d.mwu.CreateOrUpdateVRGRolesManifestWork(homeCluster); err != nil {
+		d.log.Error(err, "failed to create or update VolumeReplicationGroup Roles manifest")
 
-	if mw != nil {
-		primary, err := d.isVRGPrimary(mw)
-		if err != nil {
-			d.log.Error(err, "Failed to check whether the VRG is primary or not", "mw", mw.Name)
+		return fmt.Errorf("failed to create or update VolumeReplicationGroup Roles manifest in namespace %s (%w)",
+			homeCluster, err)
+	}
 
-			return err
-		}
+	if err := d.mwu.CreateOrUpdateVRGManifestWork(
+		d.instance.Name, d.instance.Namespace,
+		homeCluster, d.drPolicy,
+		d.instance.Spec.PVCSelector); err != nil {
+		d.log.Error(err, "failed to create or update VolumeReplicationGroup manifest")
 
-		if primary {
-			// Found MW and the VRG already a primary
-			return nil
-		}
+		return fmt.Errorf("failed to create or update VolumeReplicationGroup manifest in namespace %s (%w)", homeCluster, err)
 	}
 
-	// VRG ManifestWork does not exist, create it.
-	return d.processVRGManifestWork(homeCluster)
+	return nil
 }
 
-func (d *DRPCInstance) vrgManifestWorkExists(homeCluster string) (*ocmworkv1.ManifestWork, error) {
-	if d.instance.Status.PreferredDecision == (plrv1.PlacementDecision{}) {
-		return nil, nil
+func (d *DRPCInstance) updateVRGManifestWork(homeCluster string) error {
+	if err := d.mwu.CreateOrUpdateVRGManifestWork(
+		d.instance.Name, d.instance.Namespace,
+		homeCluster, d.drPolicy,
+		d.instance.Spec.PVCSelector); err != nil {
+		d.log.Error(err, "failed to update VolumeReplicationGroup manifest")
+
+		return fmt.Errorf("failed to update VolumeReplicationGroup manifest in namespace %s (%w)", homeCluster, err)
 	}
 
-	mwName := d.mwu.BuildManifestWorkName(rmnutil.MWTypeVRG)
+	return nil
+}
 
-	mw, err := d.mwu.FindManifestWork(mwName, homeCluster)
+func (d *DRPCInstance) isVRGPrimary(vrg *rmn.VolumeReplicationGroup) bool {
+	return (vrg.Spec.ReplicationState == rmn.Primary)
+}
+
+func (d *DRPCInstance) checkPVsHaveBeenRestored(homeCluster string) (bool, error) {
+	d.log.Info("Checking whether PVs have been restored", "cluster", homeCluster)
+
+	vrg, err := d.reconciler.getVRGFromManagedCluster(d.instance.Name, d.instance.Namespace, homeCluster)
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			d.log.Error(err, "failed to find ManifestWork")
+		d.log.Info("Failed to get VRG through MCV", "error", err)
 
-			return nil, fmt.Errorf("failed to find ManifestWork %s (%w)", mwName, err)
-		}
+		return false, err
 	}
 
-	if mw == nil {
-		d.log.Info(fmt.Sprintf("Manifestwork %s does not exist for cluster %s", mwName, homeCluster))
+	vrgCondition := findCondition(vrg.Status.Conditions, PVConditionMetadataAvailable)
+	if vrgCondition == nil {
+		d.log.Info("Waiting for PVs to be restored", "cluster", homeCluster)
 
-		return nil, nil
+		return false, nil
 	}
 
-	d.log.Info(fmt.Sprintf("Manifestwork %s exists (%v)", mw.Name, mw))
-
-	return mw, nil
+	return vrgCondition.Status == metav1.ConditionTrue, nil
```

Comment:
(note) for clarity we need to check generation as well. Currently as noted elsewhere we progress on generations as primary-gen1 -> secondary-gen2 -> deleted, so this should not be an issue for now.